src/movescount/movescountjson.cpp

```diff
@@ -964,15 +968,17 @@ bool MovesCountJSON::writePeriodicSample(ambit_log_sample_t *sample, QVariantMap
             }
             break;
         case ambit_log_sample_periodic_type_verticalspeed:
-            output.insert("VerticalSpeed", (double)value->u.verticalspeed/100.0);
+            if ((value->u.verticalspeed/100.0) >= -59 && (value->u.verticalspeed/100.0) <= 59){
+                output.insert("VerticalSpeed", (double)value->u.verticalspeed/100.0);
+            }
             break;
         case ambit_log_sample_periodic_type_cadence:
-            if (value->u.cadence != 0xff) {
+            if (value->u.cadence != 0xff && value->u.cadence <= 1000) {
                 output.insert("Cadence", value->u.cadence);
             }
             break;
         case ambit_log_sample_periodic_type_bikepower:
-            if (value->u.bikepower != 0xffff) {
+            if (value->u.bikepower != 0xffff && value->u.bikepower <= 2000) {
```

Comment:
CLion reports a reduntant check on the comparison to 0xff..

Reply:
In this case the idea is to check if bikepower is present and if it is, include it only if it's lower than 2000. Sometimes I get >= 3000 values.